distributed/comm/utils.py

```diff
@@ -18,6 +18,16 @@
     OFFLOAD_THRESHOLD = parse_bytes(OFFLOAD_THRESHOLD)
 
 
+# Find the function, `host_array()`, to use when allocating new host arrays
+try:
+    # Use NumPy, when available, to avoid memory initialization cost
+    import numpy
+
+    host_array = lambda n: memoryview(numpy.empty((n,), dtype="u1"))  # type: ignore
+except ImportError:
+    host_array = lambda n: memoryview(bytearray(n))  # type: ignore
```

Comment:
Nit: would prefer full function definitions. For readability and the typechecker, it's especially nice to specify that the function returns a `memoryview`.

Reply:
AFAICT `typing` doesn't handle this well ( https://github.com/python/typing/issues/593 ). So would rather skip it

Reply:
I see. Personally, I'd still prefer
```python
def host_array(n: int) -> memoryview:
    return memoryview(numpy.empty((n,), dtype="u1"))  # type: ignore
```
so that the return type of `host_array` is still tracked (and documented) as a memoryview, even if typecheckers can't validate that it actually does return a memoryview.

Reply:
Ok have converted these into functions with annotations (ignored as needed)